src/main/java/seedu/address/ui/BrowserPanel.java

```diff
@@ -104,14 +104,22 @@ public void freeResources() {
         browser = null;
     }
 
+    //@@author sarahnzx
     @Subscribe
     private void handlePersonPanelSelectionChangedEvent(PersonPanelSelectionChangedEvent event) {
         logger.info(LogsCenter.getEventHandlingLogMessage(event));
         ReadOnlyPerson person = event.getNewSelection().person;
+        String requestedSocialType = event.getSocialType();
         Person p = new Person(person);
         Iterator<SocialInfo> iterator = p.getSocialInfos().iterator();
         if (iterator.hasNext()) {
+            // if there is SocialInfo stored
             SocialInfo social = iterator.next();
+            String socialType = social.getSocialType();
+            while (!socialType.equals(requestedSocialType) && iterator.hasNext() && requestedSocialType != null) {
+                // if no social type is specified, the default social type shown will be Instagram
```

Comment:
Is this comment still valid? Seems like we are choosing it based on the order in the iterator.